source/ota.c

```diff
@@ -2968,14 +2989,21 @@ void OTA_EventProcessingTask( void * pUnused )
 {
     ( void ) pUnused;
 
-    /*
-     * OTA Agent is ready to receive and process events so update the state to ready.
-     */
-    otaAgent.state = OtaAgentStateReady;
-
-    while( otaAgent.state != OtaAgentStateStopped )
+    if( otaAgent.state == OtaAgentStateStopped )
```

Comment:
Does this change the behavior of the existing OTA state machine? As per the current code, we can restart the OTA task after its shutdown?

Reply:
No, this doesn't change the behavior of the existing OTA state. The OTA can be re-started after shutdown by calling OTA_Init() -> OTA_EventProcessingTask().